docs/wknml.md

```diff
@@ -197,3 +229,119 @@ Writes an NML object to a file on disk.
   wknml.write_nml(f, nml)
   ```
 
+<a name="wknml.nml_generation"></a>
+# wknml.nml\_generation
+
+<a name="wknml.nml_generation.random_color_rgba"></a>
+#### random\_color\_rgba
+
+```python
+random_color_rgba() -> Tuple[float, float, float, float]
+```
+
+A utility to generate a new random RGBA color.
+
+<a name="wknml.nml_generation.discard_children_hierarchy"></a>
+#### discard\_children\_hierarchy
+
+```python
+discard_children_hierarchy(groups: List[Group]) -> List[Group]
+```
+
+A utility to flatten the group structure. All sub-groups will become top-level items.
+
+<a name="wknml.nml_generation.globalize_tree_ids"></a>
+#### globalize\_tree\_ids
+
+```python
+globalize_tree_ids(group_dict: Dict[str, List[nx.Graph]])
+```
+
+A utility to in-place re-assign new and globally unqiue IDs to all Tree objects. Starts with ID 1
+
+**Arguments**:
+
+- `group_dict` _Dict[str, List[nx.Graph]]_ - A mapping of group names to a list of tree as NetworkX graph objects
+
+<a name="wknml.nml_generation.globalize_node_ids"></a>
+#### globalize\_node\_ids
+
+```python
+globalize_node_ids(group_dict: Dict[str, List[nx.Graph]])
+```
+
+A utility to in-place re-assign new and globally unqiue IDs to all Node objects. Edges are updated accordingly. Starts with ID 1.
+
+Note: Does not update any `Comment`s or `BranchPoint`s referencing these nodes.
```

Comment:
In this state (where the trees are nx.Graphs), branchpoints and comments are indicated by a `branchpoint` or `comment` attribute at the respective node, so there is no need to update any IDs. The separate branchpoints and comments lists are only generated from the nx.Graphs, later.